cpp/src/fil/fil.cu

```diff
@@ -222,14 +240,17 @@ struct forest {
 
   int num_trees_ = 0;
   int depth_ = 0;
-  int num_cols_ = 0;
   algo_t algo_ = algo_t::NAIVE;
-  int max_shm_ = 0;
   output_t output_ = output_t::RAW;
   float threshold_ = 0.5;
   float global_bias_ = 0;
   leaf_algo_t leaf_algo_ = leaf_algo_t::FLOAT_UNARY_BINARY;
-  int num_classes_ = 1;
+  shmem_size_params ssp_ = {.max_shm = 0,
```

Comment:
This would be better in the default constructor or default field values for `shmem_size_params`.